src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy

```diff
@@ -23,6 +24,54 @@ class ConfigurationFile extends ProjectFile {
         }
     }
 
+    Boolean doesExists() {
+        return exists
+    }
+
+    List<MuleComponent> findChildComponents(Map<String, String> elements, Boolean include) {
+        List<MuleComponent> componentList = []
+        def childNodes = configXml.childNodes()
+        def comps = []
+        childNodes.each {
+            node ->
+            if (include) {
+                if (checkElementExists( node, elements )) {
+                    comps.add(node)
+                }
+            } else {
+                if (!checkElementExists( node, elements )) {
+                    comps.add(node)
+                }
+            }
+        }
+
+        comps.each { comp ->
+            Map<String, String> atts = [:]
+            comp.attributes.each {
+                atts.put(it.key, it.value)
+            }
+            componentList.add(new MuleComponent(comp.name(), comp.namespaceURI(), atts))
+        }
+        return componentList
+    }
+
+    List<MuleComponent> findGlobalConfigs(Map<String, String> noneGlobalElements) {
```

Comment:
This method should by default ignore flow and sub-flow and not really be configurable.  findGlobalConfigs describes a specific functionality and if we allow it to be customized its not really doing that function anymore.  If we want to ignore additional globals from the list that shouldn't be done here.

Reply:
removed parameter. but that made this method useless I had to call different method from rules since its configurable from rule.